tests/binary/binaries_test.go

```diff
@@ -39,46 +39,84 @@ func TestMain(m *testing.M) {
 }
 
 func TestVersion(t *testing.T) {
-	if *releasedir == "" {
-		t.Skip("release dir not set")
-	}
-	binariesToTest := strings.Split(*binaries, " ")
-	if len(binariesToTest) == 0 {
-		t.Fatal("No binaries to test. Pass the --binaries flag.")
-	}
-	for _, b := range binariesToTest {
-		cmd := path.Join(*releasedir, b)
-		t.Run(b, func(t *testing.T) {
-			if b == "istiod" {
-				// Currently istiod is not using CLI flags - version/etc will be included in a
-				// detached file.
-				return
-			}
+	runBinariesTest(t, func(t *testing.T, name string) {
```

Comment:
the git diff here is pretty horrible since I did some small refactoring. the main change is just a new test, other logic is untouched.